### PR TITLE
Enable numbagg for reductions

### PR DESCRIPTION
**Pull request by dcherian**
- [ ] Tests added - check bottleneck tests
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/core/nputils.py [hunk 1]
```diff
@@ -18,11 +18,20 @@
...
+    _BOTTLENECK_AVAILABLE = False
+
+try:
+    import numbagg
+
+    _NUMBAGG_AVAILABLE = True
+except ImportError:
+    # use numpy methods instead
+    numbagg = np
+    _NUMBAGG_AVAILABLE = False
```
Comment (max-sixty) [inline xarray/core/nputils.py:34]: FYI there's a version of this at https://github.com/pydata/xarray/blob/dafd726c36e24ac77427513a4a149a6933353b66/xarray/core/rolling_exp.py#L16-L17, which has the advantage that it contains the version, which we likely want to filter on...

(though I'm not sure my approach is that good! Open to feedback...)

**Review by max-sixty — APPROVED** [review]
Awesome, thanks @dcherian !

Are there any other functions that would be helpful?

Do we need to capitulate on the `ddof`?

**Review by andersy005 — APPROVED** [review]

**Comment by dcherian** [comment]
Waiting on https://github.com/numbagg/numbagg/pull/162 to be released so I can add a version check and only defer to numbagg when `ddof=1` (instead of `0` currently)

**Comment by max-sixty** [comment]
> Waiting on [numbagg/numbagg#162](https://github.com/numbagg/numbagg/pull/162) to be released so I can add a version check and only defer to numbagg when `ddof=1` (instead of `0` currently)

Done!

**Comment by dcherian** [comment]
Should work whenever the CI environment updates.

**Review by max-sixty — APPROVED** [review]

**Pull request merged**